routing/dht/providers/providers.go

```diff
@@ -81,7 +81,7 @@ func NewProviderManager(ctx context.Context, local peer.ID, dstore ds.Datastore)
 const providersKeyPrefix = "/providers/"
 
 func mkProvKey(k key.Key) ds.Key {
-	return ds.NewKey(providersKeyPrefix + base64.StdEncoding.EncodeToString([]byte(k)))
+	return ds.NewKey(providersKeyPrefix + base64.URLEncoding.EncodeToString([]byte(k)))
 }
```

Comment:
@whyrusleeping Recommend we standardize on RawURLEncoding, that is how base64url is generally encoded.